etcdctlv3/command/util.go

```diff
@@ -47,3 +50,9 @@ func argify(s string) []string {
 	r := regexp.MustCompile("'.+'|\".+\"|\\S+")
 	return r.FindAllString(s, -1)
 }
+
+func contextWithTotalTimeout(cmd *cobra.Command) (context.Context, context.CancelFunc) {
+	var timeOut time.Duration
+	cmd.Flags().DurationVar(&timeOut, "total-timeout", 0, "")
```

Comment:
txn is low cost. Snapshot involves `watch` so we shall not add timeout to it. basically if a cmd involves watch, we need to be careful.